autowarefoundation/autoware.universe
# feat(vehicle_cmd_gate)!: support ad api

## Description

Add the pause service to vehicle cmd gate for the API to hook the vehicle start.
https://github.com/autowarefoundation/autoware/discussions/2464#discussioncomment-3698877

## Related links

https://github.com/autowarefoundation/autoware.universe/pull/1809

## Tests performed

Test with simple planning simulator.
1. Check the vehicle starts by default.
1. Check the vehicle does't start after this command.
    `ros2 service call /control/vehicle_cmd_gate/set_pause tier4_control_msgs/srv/SetPause "{pause: true}"`
1. Check this topic is true when engaged.
    `ros2 topic echo /control/vehicle_cmd_gate/is_start_requested`
1. Check the vehicle start after this command.
    `ros2 service call /control/vehicle_cmd_gate/set_pause tier4_control_msgs/srv/SetPause "{pause: false}"`
1. Check this topic is false after vehicle stops.
    `ros2 topic echo /control/vehicle_cmd_gate/is_start_requested`

## Notes for reviewers



## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
